source/_integrations/shelly.markdown

```diff
@@ -301,9 +312,36 @@ Please check from the device Web UI that the configured server is reachable.
 - Before set up, battery-powered devices must be woken up by pressing the button on the device.
 - OTA update service does not support battery-powered devices
 - Frequency sensors for generation 2 energy monitoring devices are supported by firmware 1.0.0 beta 6 or later.
+
+### Troubleshooting first generation Shelly devices raising a HA repair issue
+
+The following ideas might help you diagnose and fix the problem of the gen. 1 Shelly device CoIoT push updates not propagating to HA, and causing HA to display a repair issue:
+
+- If Shelly devices are in a different subnet than the Home Assistant server, please confirm that there is communication on `UDP` port `5683` between these subnets. Connect a computer to the same WiFi SSID at the same location / access point as the affected Shelly device(s) and try:
+  - On Windows command line: `portqry.exe -n Your_Home_Assistant_local_IP_address -p udp -e 5683 -sp 5683` This tool is available at https://www.microsoft.com/en-us/download/details.aspx?id=17148 or its GUI version PortQryUI at https://www.microsoft.com/en-us/download/details.aspx?id=24009&wa=wsignin1.0
+  - On Linux: `nc -vz -u Your_Home_Assistant_local_IP_address 5683`
+- If Home Assistant is running in a virtual machine or as a service on an operating system other than Home Assistant OS, open `UDP` port `5683` on the host device's firewall, and/or ensure that communication from this port is forwarded to the Home Assistant service.
+- Check WiFi connection quality. In the device's web UI, go to **Settings** >> **DEVICE INFO**. Confirm the proper SSID name and signal strength; if device reports WiFi RSSI of -70 dBm or a greater negative value, signal strength may be the issue. You can test this hypothesis by temporarily relocating the affected Shelly device(s) an/or their WiFi access point(s) to be closer together and/or with fewer metal obstacles between them, so that you get the shelly device(s) report the WiFi RSSI in the -40s...low 50s dBm. If the issue goes away then, one or more of the following measures may help:
+  - Optimize WiFi access point placement to boost the worst affected devices' Wi-Fi signal. Minimizing metal obstructions (includig wires) in the immediate vicinity of Wi-Fi antennae can often make a meaningful difference.
+  - If your affected Shelly devices are installed in metal electrical boxes, **only if local electrical codes allow**, consider having a qualified electrician replace any unused metal electrical box knock-outs with NM (non-metallic) push-in connectors. This will allow  additional Wi-Fi signal pathways to the Shelly device.
+  - If your affected Shelly devices have a dense mass of electrical wiring surrounding them, consider having a qualified electrician clean up the wiring. Multi-conductor lever connectors, such as WAGO 221-415 *(example - not endorsement)*, can allow for more compact connections compared to traditional twist nuts.
```

Comment:
I'm not sure we need this. This is HA integration documentation, not an electrical manual.

Reply:
This is an excellent point to ponder:

- On the one hand, the reality is,
  - many Shelly switches are situated in electrical boxes whose location and material are governed by local electrical codes, 
  - users do report very, very low signal strengths / poor connections, 
  - a very poor connection is something known to create problems.
  
This hurts the entire HA ecosystem by spoiling the user experience and adding support overhead - hence dampening HA's vitality and spread. Unmitigated poor connections (where it could be fixed with the right knowledge) belong in a lab. So whether a firewall, a subnet, a VLAN, or a metal electrical box are blocking the communication, the end result is the same. Hence, awareness ought to be the same. Not everyone has RF communications knowledge, and most electricians simply treat smart switches as just regular switches a customer provided. HA user awareness and directions at install time along the lines outlined in the PR can significantly affect the outcomes.
- On the other hand, the same RF propagation and signal strength issues can and do afflict not just Shelly but many other devices, such as Z-Wave, Zigbee, and a plethora of other wireless protocols. Hence:
  - While I believe firmly that this awareness and advice (expanding on basics already in the version currently live and offering additional options where, e.g. the current advice to relocate the electrical box with a Shelly device may be quite impractical for many installations),
  - I do agree with your point that this RF knowledge and solution ideas belongs somewhere other than Shelly integration documentation.

I'll therefore trim this section to a bare minimum and will look for a place to open a separate PR that will add this knowledge that can be linked from any integrations relying on wireless communication.